crates/sozo/walnut/src/verification.rs

```diff
@@ -3,114 +3,61 @@ use std::io;
-/// bytecode on the network, and if they are equal, it will store the source code and associate it
-/// with the class hash.
-pub async fn walnut_verify_migration_strategy(
-    ws: &Workspace<'_>,
-    rpc_url: String,
-    world_diff: &WorldDiff,
-) -> anyhow::Result<()> {
-    let ui = ws.config().ui();
-    // Check if rpc_url is localhost
-    if rpc_url.contains("localhost") || rpc_url.contains("127.0.0.1") {
-        ui.print(" ");
-        ui.warn("Verifying classes with Walnut is only supported on hosted networks.");
-        ui.print(" ");
-        return Ok(());
-    }
-
-    // Check if there are any contracts or models in the strategy
-    if world_diff.is_synced() {
-        ui.print(" ");
-        ui.print("🌰 No contracts or models to verify.");
-        ui.print(" ");
-        return Ok(());
-    }
+#[derive(Debug, Serialize)]
+struct VerificationPayload {
+    /// JSON that contains a map where the key is the path to the file and the value is the content
+    /// of the file. It should contain all files required to build the Dojo project with Sozo.
+    pub source_code: Value,
 
-    let _profile_config = ws.load_profile_config()?;
+    pub cairo_version: String,
+}
 
-    for (_selector, resource) in world_diff.resources.iter() {
-        if resource.resource_type() == ResourceType::Contract {
-            match resource {
-                ResourceDiff::Created(ResourceLocal::Contract(_contract)) => {
-                    // Need to verify created.
-                }
-                ResourceDiff::Updated(_, ResourceRemote::Contract(_contract)) => {
-                    // Need to verify updated.
-                }
-                _ => {
-                    // Synced, we don't need to verify.
-                }
-            }
-        }
-    }
+/// Verifies all classes in the workspace.
+///
+/// This function verifies all contracts and models in the workspace. It sends a single request to
+/// the Walnut backend with the source code. Walnut will then build the project and store
+/// the source code associated with the class hashes.
+pub async fn walnut_verify(ws: &Workspace<'_>) -> anyhow::Result<()> {
+    let ui = ws.config().ui();
 
     // Notify start of verification
     ui.print(" ");
     ui.print("🌰 Verifying classes with Walnut...");
     ui.print(" ");
 
     // Retrieve the API key and URL from environment variables
-    let _api_key = walnut_get_api_key()?;
-    let _api_url = walnut_get_api_url();
+    let api_url = walnut_get_api_url();
 
-    // Collect source code
-    // TODO: now it's the same output as scarb, need to update the dojo fork to output the source
-    // code, or does scarb supports it already?
+    // its path to a file so `parent` should never return `None`
+    let root_dir: &Path = ws.manifest_path().parent().unwrap().as_std_path();
 
-    Ok(())
-}
+    let source_code = _collect_source_code(root_dir)?;
```

Comment:
Let's remove the leading `_` since the functions are now used. Every function that is used must not be prefixed by `_`.